src/MarieAssembler.c

```diff
@@ -1010,66 +996,51 @@ wchar_t *LoadFileIntoMemory(wchar_t *FileName, int *Success) {
-		const wchar_t *HelpMessage = 
-			L"* Usage: %s <Path To Source> [Output File Name]\n"
-			"* This program compiles Marie Assembly code into logisim rom images.\n"
-			"* <Path To Source> is the path to the assembly code you'd like to assemble.\n"
-			"* [Output File Name] is the name of the files that this program will emit.\n"
-			"** If [Output File Name] is not provied, then this program generates [Output File Name] in the form of OUT_<File Name>.\n"
-			"* For Every compliation:\n"
-			"** A Raw Image file will be emitted as [Output File Name].hex\n"
-			"** A listing file will be emitted as [Output File Name].lst\n"
-			"** A symbol table will be output as [Output File Name].sym\n"
-			"** A logisim image file will bit output as [Output File Name].LogisimImage\n";
-		wprintf(HelpMessage, Args[0]);
-		return 0;
-	}
-
+int ApplicationMain(FILE *InFile, int InFileSize, FILE *OutLogisim, FILE *OutRawHex, FILE *OutSymbolTable, FILE *OutListing) {
 	int Success = TRUE;
+	if (InFile == 0) {
+		Success = FALSE;
+		printf("A input file was not provided!\n");
+	}
 
-	file_state FileState = {
-		.Line = 1,
-		.Column = 0,
-	};
-	FileState.At = LoadFileIntoMemory(Args[1], &Success);
-	if (Success == FALSE) { return 0; }
+	if (Success) {
+		file_state FileState = {
+			.Line = 1,
+			.Column = 0,
+		};
+		FileState.At = LoadFileIntoMemory(InFile, InFileSize, &Success);
 
-	paged_list *IdentifierDestinationList = AllocatePagedList(sizeof(identifier_dest), 10);
-	paged_list *IdentifierSourceList = AllocatePagedList(sizeof(identifier_source), 10);
-	
-	if (Assemble(&FileState, IdentifierDestinationList, IdentifierSourceList)) {
-		wchar_t *OutputFileName;
-		int OutputNameLength;
+		paged_list *IdentifierDestinationList = AllocatePagedList(sizeof(identifier_dest), 10);
+		paged_list *IdentifierSourceList = AllocatePagedList(sizeof(identifier_source), 10);
+		
+		Success = Assemble(&FileState, IdentifierDestinationList, IdentifierSourceList);
+		
+		if (Success) {
+			if ((OutLogisim == 0) && (OutRawHex == 0) && (OutSymbolTable == 0) && (OutListing == 0)) {
+				Success = FALSE;
+				printf("Warning: No outputs were were requested. No output files are being generated.\n");
+			}
 
-		if (ArgCount == 2) {
-			OutputFileName = Platform_CreateOutputFileName(Args[1], &Success);
-		}
-		else {
-			OutputFileName = Args[2];
+			if (OutRawHex != 0) {
+				int Temp = OutputRawHex(OutRawHex);
+				Success = Success ? Temp : FALSE; // only take on the new output value 
+			}
+			if (OutLogisim != 0) {
+				int Temp = OutputLogisimImage(OutLogisim);
+				Success = Success ? Temp : FALSE;
+			}
+			if (OutSymbolTable != 0) {
+				int Temp = OutputSymbolTable(OutSymbolTable, IdentifierDestinationList, IdentifierSourceList);
+				Success = Success ? Temp : FALSE;
+			}
+			if (OutListing != 0) {
+				int Temp = OutputListing(OutListing, IdentifierDestinationList, IdentifierSourceList);
+				Success = Success ? Temp : FALSE;
+			}
```

Comment:
Little confused by this logic. As long as we pass the conditional `if ((OutLogisim == 0) && (OutRawHex == 0) && (OutSymbolTable == 0) && (OutListing == 0))` stated on line 1019, we're guaranteed to have true values in our success variables as I don't see any error checking or handling in any of the methods for outputting the desired files. Is there any reason to check for the return values of each method, or are you planning on doing some checking in the future?

Reply:
On line 1045 we return the value of Success.

Reply:
Yeah, but at that point we could just set Success to true before line 1024 instead of having to include 4 additional lines that would amount to the same thing. If your functions are guaranteed to work, then why check their value for success?

Reply:
who said anything about those functions being guaranteed to work?
wait... did you see that all the `Output*` functions end with `return TRUE; // @TODO @NoMerge` and still merge this???

Reply:
Yeah I thought you wanted a review before you merged into master. I saw that in the source but assumed that you still wanted the merge.

Use [this](https://github.blog/2019-02-14-introducing-draft-pull-requests/) in the future to prevent this sort of issue from occuring.

Reply:
If you reviewed this and had unanswered critiques of it, why did you merge it regardless!

Reply:
Michael, I had trusted that you knew what you were doing, the critiques I had were mostly stylistic.

Also, you don't put WIP comments in your code you use the OFFICIAL feature on GitHub to convey the intent that a pull request is WIP review. I already put in another pull request to revert; we have version control to undo mistakes like these.

Reply:
> I had trusted that you knew what you were doing

You clearly need to have more faith in yourself, if I'm asking for your opinion on something, it's because I want your opinion on it. If you had problems/opinions with the code, then clearly the code I though was good to merge was inface NOT good to merge.

> use the OFFICIAL feature on GitHub to convey the intent that a pull request is WIP review.

I don't think you understand what happened here.
I thought my job was done, and it turned out that it wasn't. The reason I requested a review was for someone else to make sure I did my job, and I did it correctly. 

Most importantly, you saw "@NoMerge" in the source code, told me that "[The code] Looks good!", and merged it! I don't know how that happened without raising some red flags on your end!